Add fixes for Ubuntu 16.04:

giuseppecossu opened the pull request:
- Add fastcgi configuration for php7
 - Add required packege libapache2-mod-php

Commit:
Add fixes for Ubuntu 16.04:
 - Add fastcgi configuration for php7
 - Add required packege libapache2-mod-php

giuseppecossu commented:
Hi,
I've tested on Ubuntu 16.04 and with the proposed fixes it works. I don't know why the CI fails, apparently is not related to my changes.

Voronenko commented:
@giuseppecossu   I will check, and address, if any

Voronenko commented:
@giuseppecossu  I propose to keep it synchronized to  https://github.com/softasap/sa-lamp/blob/master/defaults/main.yml#L25

I have this role used in quite older deployments, thus I need to support backward compability.

Thus all parts, that refer to php7, need to be aligned by  php_family_prefix (i.e. php_family_prefix == php7)

Voronenko commented on tasks/php_apache.yml:
```diff
@@ -1,8 +1,3 @@
-    when: "apache_mode == 'worker'"
-    become: yes
-    tags: setup
 
```
^^^^ you moved this below enabling the conf. 

That's why automated test fails on step - on first run you don't have template created => nothing to enable.

TASK [role_under_test : LAMP | a2enconf fastcgi] *******************************
fatal: [localhost]: FAILED! => {"changed": true, "cmd": ["a2enconf", "fastcgi"], "delta": "0:00:00.065135", "end": "2017-03-06 10:23:47.813574", "failed": true, "rc": 1, "start": "2017-03-06 10:23:47.748439", "stderr": "ERROR: Conf fastcgi does not exist!", "stdout": "", "stdout_lines": [], "warnings": []}

Requested change: order needs to stay the same

Voronenko commented on tasks/php_apache.yml:
```diff
@@ -73,6 +74,12 @@
   - block:
 
+    - name: LAMP | FAST-CGI configuration for php
+      template: src={{role_dir}}/templates/apache2/conf/fastcgi-php7.conf.j2 dest=/etc/apache2/conf-available/fastcgi.conf
```
This part needs also be moved above BEFORE fastcgi module is enabled.

Also, I (and potentially some other guys) have this role used in quite older deployments, thus we need to support backward compability.

All parts, that refer to php7, need to be aligned by php_family_prefix (i.e. php_family_prefix == php7).

Also original lines 1-5 need to be reverted back
(template: src={{role_dir}}/templates/apache2/conf/fastcgi.conf.j2 dest=/etc/apache2/conf-available/fastcgi.conf)

but this time in when they need to match to php_family_prefix php5

Voronenko commented on tasks/php_apache.yml:
```diff
@@ -126,6 +133,13 @@
         - restart apache2
 
+    - name: LAMP | Install PHP libapache 
+      apt: name=libapache2-mod-php state=latest
```
This is a bit strange, because I have clean deployments with existing role, where somehow this module is installed.
If this will play on 14.04 LTS, this should be ok, probably

giuseppecossu replied:
Ok, It seems required on 16.04

Commit:
Fix fastcgi template creation
 - Add php_family_prefix set_facts: default value in xenial is php7

Voronenko commented on tasks/php_apt_switchversion.yml:
```diff
@@ -16,13 +16,14 @@
   - name: PHP | Fix php family prefix
-    set_fact: php_family_prefix=php
+    set_fact: php_family_prefix=php7
     when: php_family == "default" and (ansible_distribution_release == "xenial") and  php_present|failed
```
and  php_present|failed can be eliminated.

Below there were PPA, that's why we were saving some time on skipping.
while here we set fact, which should be executed always

Voronenko commented on tasks/php_apt_switchversion.yml:
```diff
@@ -16,13 +16,14 @@
       -setup
 
   - name: PHP | Add apt repository for php 5.4
+    set_fact: php_family_prefix=php5
```
you can't combine multiple modules in single step (apt_repository and set_Fact) in that case.
This for 100% should get play error

Commit:
Fix set_fact wrong configuration

Commit:
Fix php configuration on xenial:
 - use fastcgi-php.conf.j2 file for php7 configuration with php prefix
 - add php_family 7.0 description in defaults

giuseppecossu commented:
I've tested on Ubuntu 16.04 and now it should be ok. I've seen that for xenial you adopt "php_family_prefix=php", so basically I use that prefix for the configuration file also.

Voronenko commented:
Thanks, added pr to a list of tasks to check/merge!

Voronenko commented on tasks/php_apache.yml:
```diff
@@ -1,5 +1,5 @@
   - name: LAMP | FAST-CGI configuration for php
-    template: src={{role_dir}}/templates/apache2/conf/fastcgi.conf.j2 dest=/etc/apache2/conf-available/fastcgi.conf
+    template: src={{role_dir}}/templates/apache2/conf/fastcgi-{{ php_family_prefix }}.conf.j2 dest=/etc/apache2/conf-available/fastcgi.conf
```
in PR template is still named  fastcgi-php.conf.j2

if you matching -php7 correction is needed

giuseppecossu replied:
If you put there php7 it doesn't work with php extension on xenial, e.g., "{{php_family_prefix}}-mysql". So I think "-php" is ok.

giuseppecossu replied:
An option could be avoid using that variable, but php_family_prefix=php still remains "php" as you specified in php_apt_switchversion.yml, right?

Voronenko replied:
Somehow did not have setup with php7 yet.  For php5 packages where called php5-*
Did they change approach with php7 to be just php-* ?

giuseppecossu replied:
On xenial you can choose php* or php7.0, but not php7.
